numpy/core/src/umath/ufunc_object.c

```diff
@@ -1730,10 +1745,154 @@ make_arr_prep_args(npy_intp nin, PyObject *args, PyObject *kwds)
     }
 }
 
+/*
+ * Check whether any of the outputs of a gufunc has core dimensions.
+ */
+static int
+_has_output_coredims(PyUFuncObject *ufunc) {
+    int i;
+    for (i = ufunc->nin; i < ufunc->nin + ufunc->nout; ++i) {
+        if (ufunc->core_num_dims[i] > 0) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+/*
+ * Interpret a possible axes keyword argument, using it to remap the default
+ * axes (at the end of the operands) to new ones.
+ *
+ * Returns 0 on success, and -1 on failure
+ */
+static int
+_parse_axes_arg(PyUFuncObject *ufunc, PyObject *axes, PyArrayObject **op,
+                int broadcast_ndim, int **remap_axis) {
+    int nin = ufunc->nin;
+    int nout = ufunc->nout;
+    int nop = nin + nout;
+    int op_ndim, core_ndim, num_axes;
+    int i, j, iter_axis;
+    int requested_axis = 0;
+    int list_size = 0;
+    int item_is_tuple = 0;
```

Comment:
Move `item_is_tuple`, `num_axes`, `op_ndim`, `core_ndim`, `iter_axis`, `requested_axis`, `j` within the loop.